packages/nextjs-mf/src/plugins/container/InvertedContainerRuntimeModule.ts

```diff
@@ -64,6 +72,547 @@ class InvertedContainerRuntimeModule extends RuntimeModule {
+        ]
+      )});`,
+      `var loadStrictVersionCheckFallback = /*#__PURE__*/ init(${runtimeTemplate.basicFunction(
+        'scopeName, scope, key, version, fallback',
+        [
+          `var entry = scope && ${RuntimeGlobals.hasOwnProperty}(scope, key) && findValidVersion(scope, key, version);`,
+          `return entry ? get(entry) : fallback();`,
+        ]
+      )});`,
+      `var loadStrictSingletonVersionCheckFallback = /*#__PURE__*/ init(${runtimeTemplate.basicFunction(
+        'scopeName, scope, key, version, fallback',
+        [
+          `if(!scope || !${RuntimeGlobals.hasOwnProperty}(scope, key)) return fallback();`,
+          'return getStrictSingletonVersion(scope, scopeName, key, version);',
+        ]
+      )});`,
+      'var installedModules = __webpack_require__.installedModules',
+      'var moduleToHandlerMapping = {',
+      Template.indent(
+        Array.from(
+          moduleIdToSourceMapping,
+          ([key, source]) => `${JSON.stringify(key)}: ${source.source()}`
+        ).join(',\n')
+      ),
+      '};',
+      'var listOfInitialIds = {',
+      Template.indent(
+        Array.from(moduleIdToSourceChunkID, ([key, chunkIds]) => {
+          return `${JSON.stringify(key)}: ${JSON.stringify(
+            Object.keys(chunkIds)
+          )}`;
+        }).join(',\n')
+      ),
+      '};',
+
+      initialConsumes.length > 0
+        ? Template.asString([
+            //@ts-ignore
+            `var initialConsumes = ${JSON.stringify(initialConsumes)};`,
+            // `initialConsumes.forEach(${runtimeTemplate.basicFunction('id', [
+            //   `${
+            //     RuntimeGlobals.moduleFactories
+            //   }[id] = ${runtimeTemplate.basicFunction('module', [
+            //     '// Handle case when module is used sync',
+            //     'installedModules[id] = 0;',
+            //     `delete ${RuntimeGlobals.moduleCache}[id];`,
+            //     'var factory = moduleToHandlerMapping[id]();',
+            //     'if(typeof factory !== "function") throw new Error("Shared module is not available for eager consumption: " + id);',
+            //     `module.exports = factory();`,
+            //   ])}`,
+            // ])});`,
+          ])
+        : '// no consumes in initial chunks',
+      // this._runtimeRequirements.has(RuntimeGlobals.ensureChunkHandlers)
+      Template.asString([
+        `var chunkMapping = ${JSON.stringify(chunkToModuleMapping, null, '')};`,
+        `__webpack_require__.getEagerSharedForChunkId = ${runtimeTemplate.basicFunction(
+          'chunkId, promises',
+          [
+            `if(${RuntimeGlobals.hasOwnProperty}(chunkMapping, chunkId)) {`,
+            Template.indent([
+              `chunkMapping[chunkId].forEach(${runtimeTemplate.basicFunction(
+                'id',
+                [
+                  `if(${RuntimeGlobals.hasOwnProperty}(installedModules, id)) return promises.push(installedModules[id]);`,
```

Comment:
@alexander-akait do you have any ideas here>

When hmr happenes it seems to reset share scope and treplace remote import factories with "console.warn(hmr unexpected require)" 

Im trying to prevent it here but it seems hit and miss.